### ISSUE TITLE
Question about retrieving binary weights from the model

### ISSUE THREAD

**Issue by utexasnew** [larq/larq#685]
After building out the simple BNN from the following guide: https://docs.larq.dev/larq/tutorials/binarynet_cifar10/

I try retrieving binary weights to examine via https://docs.larq.dev/larq/guides/bnn-optimization/#retrieving-the-binary-weights

And I notice that despite the kernel quantization, I receive output that is not entirely +1 and -1. For example, values such as 8.16999555e-01 and 3.77580225e-02 appear within the weight kernel. 

Is there any intuitive explanation for this? Thank you!

**Comment by lgeiger** [comment]
Outside the `quantize_context` this is expected when training models with latent weights as explained in the docs you linked above since the weights are only binarized in the forward pass and stored as floating point values during training.

However, the following should return binarized weights for the binary convolutions:
```python
with larq.context.quantized_scope(True):
    weights = model.get_weights()  # get binary weights
```
Although keep in mind that the model might include some full precision layers like batch norms that won't appear quantized in the Keras model but can be fused when deploying with larq compute engine.

**Comment by mervess** [comment]
If the model was saved _within the scope_ beforehand, the one below also works for me when loading the model or the weights.
```
import larq as lq    # this line is necessary now when loading the model even though lq won't be used.
weights = model.get_weights()
```
